apps/web/views/org/components/NewOrgDialog.tsx

```diff
@@ -8,33 +9,86 @@ import {
 } from '@app/ui/components/dialog';
 import { Button } from '@app/ui/components/button';
 import { Input } from '@app/ui/components/input';
+import { ENV, apiHandler } from '@app/config';
+import * as yup from 'yup';
+
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useAddOrg } from '@app/hooks/api/org';
 
 type IModal = {
     isOpen: boolean;
     onClose: () => void;
 };
 
+const Schema = yup.object().shape({
+    name: yup.string().required(),
+    slug: yup.string().test("checkslug", "a url with same name already exist", (val) => {
+        return new Promise((resolve, _reject) => {
+            apiHandler.get(`/org/find/${val}`).then((el) => {
+                if (!el.data.ok) {
+                    resolve(true)
+                }
+                throw new Error()
+
+                // TODO @sreehari2003
+                // add debouncing here
+            }).catch(() => {
+                // already exist
+                resolve(false)
+            })
+        })
```

Comment:
api is called even when i click on the name input need a help here

Reply:
User don't have to enter a slug when creating org. Instead autogenarate using fakersj
They can update slug after creation